canonical-web-and-design/maas-ui
# ✨ (legacy): Add cloud-init support to legacy.

## Done

* Add support for cloud-init to deploy form on machine details view.
* Fix styles to make deploy form better aligned with the react version on machine list.

*Note*: the **actual** diff is not that large, `node-details.html` has been reformatted here (curious that prettier missed that earlier...). Apologies for the heart attack.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

from machine details:
- Select "Take action" ... "Deploy"
- Tick "Cloud-init user-data…"
- Add a cloud config file via the file picker from "Customise options", it should populate the text area.
- Submit the form with your network inspector open, the deploy payload should contain extra data `user_data`.

## Fixes

Fixes: canonical-web-and-design/MAAS-squad#2041

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
